packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -105,23 +107,20 @@ - (dispatch_queue_t)methodQueue
 - (void)startObserving {
     _hasListeners = YES;
     
-    // Note that [UITraitCollection currentTraitCollection] always returns the same default trait collection,
-    // presumably because FRNAppearanceAdditions isn't a view, so it never gets updated with the right traitCollection
-    // (which happens when a view gets added to the view hierachy). In order to get the right trait collection,
-    // we need to access a view that's been added to the view hierarchy
-    UIViewController *viewControllerWithInitialTraitCollection = RCTPresentedViewController();
-    UITraitCollection *initialTraitCollection;
-    
-    if (viewControllerWithInitialTraitCollection != nil) {
-        initialTraitCollection = [viewControllerWithInitialTraitCollection traitCollection];
-    } else {
-        initialTraitCollection = [UITraitCollection currentTraitCollection];
+    for (UIWindow *window in RCTSharedApplication().windows) {
+        RCTRootView *rootView = (RCTRootView *)[[window rootViewController] view];
+        
+        if ([rootView isKindOfClass:[RCTRootView class]]) {
+            if ([[rootView bridge] bundleURL] == [[self bridge] bundleURL]) {
```

Comment:
The bundleURL isn't the same across windows?

Reply:
Can we literally check `if ([[rootView bridge] isEqual:[self bridge]])` ?

Reply:
So at runtime, `[rootView bridge]` is an RCTBridge, but [self bridge] is an RCTCxxBridge, so that check alone always fails. I looked up what RCTCxxBridge is - it gets initialized with a parent RCTBridge. I had to import <React/RCTBridge+Private.h> in order to be able to access the parent bridge, which I'm not sure if I should do, but this works the same as before.